DependencyInjection/DoctrineExtension.php

```diff
@@ -886,11 +886,21 @@ protected function getMappingResourceExtension(): string
     protected function loadCacheDriver($cacheName, $objectManagerName, array $cacheDriver, ContainerBuilder $container, bool $usePsr6 = false): string
     {
         $aliasId = $this->getObjectManagerElementName(sprintf('%s_%s', $objectManagerName, $cacheName));
-        $isPsr6  = null;
 
         switch ($cacheDriver['type'] ?? 'pool') {
             case 'service':
                 $serviceId = $cacheDriver['id'];
+                $isPsr6    = $cacheDriver['is_psr6'];
+
+                if (! $isPsr6) {
+                    trigger_deprecation(
```

Comment:
if you want to use the `trigger_deprecation` function of Symfony, you need to add the dependency on `symfony/deprecation-contract` to make Psalm happy (users will probably already have it thanks to FrameworkBundle, but that's quite fragile)

Reply:
I thought I had added it already, but it's there now.